### PR TITLE
fix(tests): fix #149 update tests for already existing publishers, with fixtures

### LINKED ISSUE #149: Change tests and migrate tests that checks for already existing publishers to fixtures
Looks like there's something wrong with tests for publishers' description. We should use fixtures as a base to tell which record are present, and not previous tests.

_Originally posted by @bfabio in https://github.com/italia/developers-italia-api/issues/148#issuecomment-1261044734_

### PR DESCRIPTION
**Pull request by LorenzoS92**
Fix #149 update tests for already existing publishers, with fixtures

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -527,7 +527,7 @@ func TestPublishersEndpoints(t *testing.T) {
 		},
 		{
 			query: "POST /v1/publishers - Email already exist",
-			body:  `{"codeHosting": [{"url" : "https://example-testcase-xx3.com"}], "email":"example-testcase-3@example.com", "externalCode":"example-testcase-3", "description": "test"}`,
+			body:  `{"codeHosting": [{"url" : "https://example-testcase-xx3.com"}], "email":"foobar@1.example.org", "externalCode":"example-testcase-3", "description": "test"}`,
```
Comment (bfabio) [inline main_test.go:530]:
```suggestion
			body:  `{"codeHosting": [{"url" : "https://example-testcase-xx3.com"}], "email":"foobar@1.example.org", "description": "test"}`,
```

File: main_test.go [hunk 2]
```diff
@@ -538,7 +538,7 @@ func TestPublishersEndpoints(t *testing.T) {
 		},
 		{
 			query: "POST /v1/publishers - Description already exist",
-			body:  `{"codeHosting": [{"url" : "https://example-testcase-xx3.com"}], "email":"example-testcase-3-unique@example.com", "externalCode":"example-testcase-3", "description": "test"}`,
+			body:  `{"codeHosting": [{"url" : "https://example-testcase-xx3.com"}], "email":"example-testcase-3-unique@example.com", "externalCode":"example-testcase-3", "description": "Publisher description 1"}`,
```
Comment (bfabio) [inline main_test.go:541]:
```suggestion
			body:  `{"codeHosting": [{"url" : "https://example-testcase-xx3.com"}], "email":"example-testcase-3-unique@example.com", "description": "Publisher description 1"}`,
```

**Review by bfabio — COMMENTED** [review]
Making the tests minimal with just the necessary fields

**Comment by LorenzoS92** [comment]
Could you try to rebase now? @bfabio

**Comment by bfabio** [comment]
@LorenzoS92 even this PR doesn't pass anymore with the suggestions merged. Something's wrong with the description test

**Comment by LorenzoS92** [comment]
@bfabio Have you changed the fixtures by any chance or something else?

File: go.mod [hunk 3]
```diff
@@ -16,7 +16,10 @@ require (
 	gorm.io/gorm v1.23.6
 )
 
-require golang.org/x/exp v0.0.0-20220827204233-334a2380cb91
+require (
+	github.com/Jeffail/gabs/v2 v2.6.1
+	golang.org/x/exp v0.0.0-20220827204233-334a2380cb91
+)
```
Comment (bfabio) [inline go.mod:22]:
```suggestion
```

File: go.sum [hunk 4]
```diff
@@ -1,4 +1,6 @@
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+github.com/Jeffail/gabs/v2 v2.6.1 h1:wwbE6nTQTwIMsMxzi6XFQQYRZ6wDc1mSdxoAN+9U4Gk=
+github.com/Jeffail/gabs/v2 v2.6.1/go.mod h1:xCn81vdHKxFUuWWAaD5jCTQDNPBMh5pPs9IJ+NcziBI=
```
Comment (bfabio) [inline go.sum:3]:
```suggestion
```

File: internal/database/database_test.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package database
```
Comment (bfabio) [inline internal/database/database_test.go:1]: Need to remove this file

**Review by bfabio — APPROVED** [review]

**Pull request merged**